### PR TITLE
Fix #200

### LINKED ISSUE #200: TypeError: Cannot read property 'prefix' of undefined in openFileDescriptor
heey! 

optional prefix for metrics is failing because config is undefined.

thank you!

```
/opt/atlassian/pipelines/agent/build/node_modules/prom-client/lib/metrics/processOpenFileDescriptors.js:13
	const namePrefix = config.prefix ? config.prefix : '';
	                          ^
TypeError: Cannot read property 'prefix' of undefined
```

### PR DESCRIPTION
**Pull request by frankie567**
Fix the issue #200. The default value for `config` was missing. I added a unit test.

I took the liberty to add Sinon in devDependencies in order to fake the platform, so that we can run the unit test on any platform ; because this metric is otherwise only available on Linux.

Regards!

### CODE REVIEW

**Comment by siimon** [comment]
Thanks!

Do you mind updating the changelog aswell?

**Comment by frankie567** [comment]
Done! Hope it matches the way you do it usually 🙂

**Comment by siimon** [comment]
Great! Thanks 👍

**Comment by siimon** [comment]
Published in 11.1.1

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
 		"lint-staged": "^7.0.0",
 		"lolex": "^2.1.3",
 		"prettier": "1.11.1",
+		"sinon": "^6.0.1",
```
Comment (SimenB) [inline package.json]: We have Jest which can do this already

Comment (frankie567) [reply package.json]: Didn't know about that, sorry. I can make another PR to fix that if you wish.

Comment (SimenB) [reply package.json]: That'd be awesome!

Comment (frankie567) [reply package.json]: I tried a bit, but I have to admit I'm not very confident with Jest. You'll find below what I tried, but without luck. As you seem to be a big contributor to Jest, I hope you can point me in the right direction :)

**Attempt 1**

`package.json`
```json
"jest": {
  "testEnvironment": "node",
  "testEnvironmentOptions": {
    "process": {
      "platform": "linux" 
    }
  },
  "testRegex": ".*Test\\.js$"
},
```

**Attempt 2**

`processOpenFileDescriptorsTest.js`
```js
jest.mock('process', () => {
  return {
    platform: 'linux'
  };
});
```

Comment (SimenB) [reply package.json]: Since all test files run in isolation in Jest, you can just assign.

```js
process.platform = 'linux';
```

**Pull request merged**